test/controllers/UpdateAdditionalInformationControllerSpec.scala

```diff
@@ -0,0 +1,179 @@
+    )
+    private lazy val dataRetrievalAction = new FakeDataRetrievalAction(userAnswers)
+
+    implicit lazy val dataRequest: DataRequest[AnyContentAsEmpty.type] = DataRequest(
+      OptionalDataRequest(
+        IdentifierRequest(fakeRequest, "credId", "eori"),
+        "credId",
+        "eori",
+        userAnswers
+      ),
+      "credId",
+      "eori",
+      userAnswers.get
+    )
+
+    val formProvider: UpdateAdditionalInformationFormProvider = injector.instanceOf[UpdateAdditionalInformationFormProvider]
+    val form: UpdateAdditionalInformationFormProvider = formProvider
+
+    MockedSessionRepository.set(Future.successful(true))
+
+    lazy val controller = new UpdateAdditionalInformationController(authenticatedAction, dataRetrievalAction, dataRequiredAction,
+      mockSessionRepository, messagesControllerComponents, form, view)
+  }
+
+  "GET onLoad" should {
+    "return OK" in new Test {
+      val result: Future[Result] = controller.onLoad(fakeRequest)
+      status(result) mustBe Status.OK
+    }
+
+    "return HTML" in new Test {
+      override val userAnswers: Option[UserAnswers] = Some(UserAnswers("some-cred-id").set(UpdateAdditionalInformationPage, "some text").success.value)
+      val result: Future[Result] = controller.onLoad(fakeRequest)
+      contentType(result) mustBe Some("text/html")
+      charset(result) mustBe Some("utf-8")
+    }
+  }
+
+  "POST onSubmit" when {
+    "payload contains valid data when check mode is false" should {
+
+      "return a SEE OTHER response" in new Test {
+        val request: FakeRequest[AnyContentAsFormUrlEncoded] = fakeRequest.withFormUrlEncodedBody("value" -> "some text")
+        lazy val result: Future[Result] = controller.onSubmit(request)
+        status(result) mustBe Status.SEE_OTHER
+      }
+
+      "return the correct location header for the response" in new Test {
+        override val userAnswers: Option[UserAnswers] = Some(
+          UserAnswers("some-cred-id")
+            .set(UpdateAdditionalInformationPage, "some text").success.value
+            .set(CheckModePage, false).success.value
+        )
+        val request: FakeRequest[AnyContentAsFormUrlEncoded] = fakeRequest.withFormUrlEncodedBody("value" -> "some text")
+        lazy val result: Future[Result] = controller.onSubmit(request)
+        redirectLocation(result) mustBe Some(controllers.routes.UpdateAdditionalInformationController.onLoad().url)
+      }
+
+      "update the UserAnswers in session" in new Test {
+        private val request = fakeRequest.withFormUrlEncodedBody("value" -> "some text")
+        await(controller.onSubmit(request))
+        verifyCalls()
+      }
+    }
+
+    "payload contains invalid data" should {
+      "return a BAD REQUEST" in new Test {
+        val result: Future[Result] = controller.onSubmit(fakeRequest)
+        status(result) mustBe Status.BAD_REQUEST
+      }
+    }
+  }
+
+  //  "payload contains valid data when check mode is true" should {
```

Comment:
Why is this commented out? If it is no longer needed then could it be deleted please?